### PR TITLE
feat: implement task system & kuai context

### PR DESCRIPTION
**Pull request by PainterPuppets**
create `kuaiproject/core` package & implements task system 

 - [x] implements task system
 - [x] create `KuaiContext` for initial kuai project
 - [x] integrate task system & cli
 - [x] unit test for task system
 - [ ] unit test for `KuaiContext`
 - [ ] unit test for `cli`

task system design: https://github.com/ckb-js/kuai/issues/13

### CODE REVIEW

**Comment by Keith-CY** [comment]
How is it going

**Comment by Keith-CY** [comment]
Please assign reviewers since the PR is marked as ready for review

**Comment by PainterPuppets** [comment]
this pr contains a lot of commits,should i split it?

**Comment by Keith-CY** [comment]
> this pr contains a lot of commits,should i split it as multi pr?

There are 1000 lines of diff, lock file excluded, so it would be good for review to split them into small PRs

File: packages/core/src/errors.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import { ErrorDescriptor, ERRORS } from './errors-list';
+
+const inspect = Symbol.for('nodejs.util.inspect.custom');
```
[the remark this answers is not in the document]
Comment (felicityin) [reply packages/core/src/errors.ts:3]: The `nodejs.util.inspect.custom` symbol has been registered globally: https://nodejs.org/api/util.html#utilinspectcustom
The Symbol.for(key) method searches for existing symbols in a runtime-wide symbol registry with the given key and returns it if found. 
So `Symbol.for('nodejs.util.inspect.custom')` always returns the same value.

File: packages/core/src/errors.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    const varTag = `%${varName}%`;
+
+    if (!template.includes(varTag)) {
+      throw new KuaiError(ERRORS.INTERNAL.TEMPLATE_VARIABLE_TAG_MISSING, {
+        var: varName,
+      });
+    }
+  }
+
+  if (template.includes('%%')) {
```
[the remark this answers is not in the document]
Comment (felicityin) [reply packages/core/src/errors.ts:94]: It's already out of the loop.

File: packages/core/src/errors.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+  for (const varName of Object.keys(values)) {
+    let varValue: string;
+
+    if (values[varName] === undefined) {
+      varValue = 'undefined';
+    } else if (values[varName] === null) {
+      varValue = 'null';
+    } else {
+      varValue = values[varName].toString();
```
[the remark this answers is not in the document]
Comment (felicityin) [reply packages/core/src/errors.ts:109]: Any object in JS has toString() method, except null and undefined.

[the remark this answers is not in the document]
Comment (felicityin) [reply packages/core/src/errors.ts:109]: `undefined` and `null` can make error reporting more friendly.

**Pull request closed without merging**